ui/src/main/js/page/settings/standalone/SettingsProxyConfiguration.js

```diff
@@ -0,0 +1,146 @@
+            updateRequest={() => ConfigurationRequestBuilder.createUpdateRequest(proxyRequestUrl, csrfToken, settingsProxyConfig.id, settingsProxyConfig)}
+            validateRequest={() => ConfigurationRequestBuilder.createValidateRequest(proxyRequestUrl, csrfToken, settingsProxyConfig)}
+            readonly={readOnly}
+            displaySave={displaySave}
+            displayTest={false}
+            displayDelete={displayDelete}
+            errorHandler={errorHandler}
+        >
+            <h2 key="settings-header">Proxy Configuration</h2>
+            <TextInput
+                id={SETTINGS_FIELD_KEYS.proxyHost}
+                name="proxyHost"
+                label="Proxy Host"
+                description="The host name of the proxy server to use."
+                readOnly={readOnly}
+                onChange={fieldModelUtilities.handleTestChange(settingsProxyConfig, setSettingsProxyConfig)}
+                value={settingsProxyConfig.proxyHost || undefined}
+                errorName="proxyHost"
+                errorValue={errors.fieldErrors.proxyHost}
+            />
+            <NumberInput
+                id={SETTINGS_FIELD_KEYS.proxyPort}
+                name="proxyPort"
+                label="Proxy Port"
+                description="The port of the proxy server to use."
+                readOnly={readOnly}
+                onChange={fieldModelUtilities.handleTestChange(settingsProxyConfig, setSettingsProxyConfig)}
+                value={settingsProxyConfig.proxyPort || undefined}
+                errorName="smtpPort"
+                errorValue={errors.fieldErrors.proxyPort}
+            />
+            <TextInput
+                id={SETTINGS_FIELD_KEYS.proxyUsername}
+                name="proxyUsername"
+                label="Proxy Username"
+                description="If the proxy server requires authentication, the username to authenticate with the proxy server."
+                readOnly={readOnly}
+                onChange={fieldModelUtilities.handleTestChange(settingsProxyConfig, setSettingsProxyConfig)}
+                value={settingsProxyConfig.proxyUsername || undefined}
+                errorName="proxyUsername"
+                errorValue={errors.fieldErrors.proxyUsername}
+            />
+            <PasswordInput
+                id={SETTINGS_FIELD_KEYS.proxyPassword}
+                name={passwordName}
+                label="Proxy Password"
+                description="If the proxy server requires authentication, the password to authenticate with the proxy server."
+                readOnly={readOnly}
+                onChange={fieldModelUtilities.handleTestChange(settingsProxyConfig, setSettingsProxyConfig)}
+                value={settingsProxyConfig[passwordName] || undefined}
+                isSet={passwordFromApiExists}
+                errorName={passwordName}
+                errorValue={errors.fieldErrors.proxyPassword}
```

Comment:
The error key will generally match the actual field name. So you should get away with errors.fieldErrors[passwordName]